### PR TITLE
feat: output paths released

### PR DESCRIPTION
**Pull request by RobertCraigie**
Needed so that we can tell what sub-packages have been released and automatically publish them.

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -9998,6 +9999,53 @@ function safeJson(input) {
...
+ */
+function setPathOutput(path, key, value) {
+  if (path === '.') {
+    core.setOutput(key, value);
+  } else {
+    core.setOutput(`${path}--${key}`, value);
+  }
+}
+
+function outputReleases(releases) {
```
Comment (dgellow) [inline dist/index.js:10015]: Could you add a docstring here to help end users understand what this function is for?

File: index.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ async function main() {
 
   if (data?.releases?.length) {
     core.setOutput('releases_created', true);
```
Comment (dgellow) [inline index.js:54]: Is this still needed? `outputReleases()` seems to also set `releases_created` to a boolean.

File: index.js [hunk 3]
```diff
@@ -66,6 +67,55 @@ function safeJson(input) {
...
+}
+
+/**
+ * Add specific outputs for each release.
+ *
+ * Most notable is the `paths_released` output which
+ * makes it easy to build automations around monorepo setups.
+ */
+function outputReleases(releases) {
+  releases = releases.filter((release) => release !== undefined);
```
Comment (dgellow) [inline index.js:90]: Should we also filter `null`, and maybe other falsey values?

```suggestion
  releases = releases.filter(Boolean);
```

We can then remove the `if (!release) continue;`.

File: index.js [hunk 4]
```diff
@@ -66,6 +67,55 @@ function safeJson(input) {
...
+
+  const pathsReleased = [];
+
+  for (const release of releases) {
+    if (!release) {
+      continue;
+    }
+
+    const path = release.path || '.';
```
Comment (dgellow) [inline index.js:99]: In which situation is the `path` unspecified? Should we log a warning if it is missing?

File: index.js [hunk 5]
```diff
@@ -66,6 +67,55 @@ function safeJson(input) {
...
+    if (path) {
```
Comment (dgellow) [inline index.js:100]: I don't think that can be false due to the `|| '.'` from the previous line.

**Review by dgellow — APPROVED** [review]
LGTM.

Not directly related to your PR, but I don't really understand why we have a `build` script for this project, it seems to be using JS directly?

**Comment by RobertCraigie** [comment]
We have a build script so that all our dependencies can be bundled into a single js file which leads to a better experience for users as they don't have to install a bunch of dependencies to use the action.

Normally we wouldn't commit the bundled code in PRs but I did so to test the action directly.

**Comment by dgellow** [comment]
I'm fine to commit the JS output, it's practical and common to do so for GitHub action in my experience

**Comment by RobertCraigie** [comment]
Ah yes sorry I meant that we have automations for committing the JS output for you.

**Pull request merged**